app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+    callbacks: PlantDetailsCallbacks,
+    transitionState: TransitionState
+) {
+    VerticalScroller(scrollerPosition) {
+        // The header transition happens given the _original_ position of the name on the screen
+        var namePosition by state { Float.MAX_VALUE }
+        // Whenever the name position or the scroller position changes,
+        // check if the toolbar should be shown or not
+        onCommit(namePosition, scrollerPosition.value) {
+            onToolbarShownUpdate(
+                scrollerPosition.value > (namePosition + HeaderTransitionOffset)
+            )
+        }
+
+        Hide(toolbarShown) { hideModifier ->
+            PlantImageHeader(
+                scrollerPosition, plant.imageUrl, callbacks.onFabClicked, isPlanted, hideModifier,
+                Modifier.drawLayer(alpha = transitionState[contentAlphaKey], clip = false)
+            )
+        }
+        PlantInformation(
+            name = plant.name,
+            wateringInterval = plant.wateringInterval,
+            description = plant.description,
+            onNamePositioned = {
+                if (namePosition == Float.MAX_VALUE) {
+                    namePosition = it.globalPosition.y
+                }
+            },
+            toolbarShown = toolbarShown
+        )
+    }
+}
+
+@Composable
+private fun PlantHeader(
+    toolbarShown: Boolean,
+    plantName: String,
+    callbacks: PlantDetailsCallbacks,
+    transitionState: TransitionState
+) {
+    if (toolbarShown) {
+        PlantDetailsToolbar(
+            plantName = plantName,
+            onBackClicked = callbacks.onBackClicked,
+            onShareClicked = callbacks.onShareClicked,
+            modifier = Modifier.drawOpacity(transitionState[toolbarAlphaKey])
+        )
+    } else {
+        PlantHeaderActions(
+            onBackClicked = callbacks.onBackClicked,
+            onShareClicked = callbacks.onShareClicked,
+            modifier = Modifier.drawOpacity(transitionState[contentAlphaKey])
+        )
+    }
+}
+
+@Composable
+private fun PlantDetailsToolbar(
+    plantName: String,
+    onBackClicked: () -> Unit,
+    onShareClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(modifier) {
+        val spacerHeight = with(DensityAmbient.current) {
+            InsetsAmbient.current.systemBars.top.toDp()
+        }
+        Spacer(
+            Modifier.preferredHeight(spacerHeight).fillMaxWidth()
+                .drawBackground(MaterialTheme.colors.surface)
+        )
+        TopAppBar(
+            backgroundColor = MaterialTheme.colors.surface
+        ) {
+            IconButton(onBackClicked, Modifier.gravity(Alignment.CenterVertically)) {
+                Icon(Icons.Filled.ArrowBack)
+            }
+            Text(
+                text = plantName,
+                color = MaterialTheme.colors.onSurface,
+                style = MaterialTheme.typography.h6,
+                modifier = Modifier.weight(1f).fillMaxSize()
```

Comment:
Do you need the `fillMaxSize`? Shouldn't it center without that? If not, perhaps comment why each needed? Also one modifier per line.

Reply:
It doesn't center: https://issuetracker.google.com/158829169

Will add comments about this, thanks!